drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/handlers/HitPolicySelectorViewImpl.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.drools.workbench.screens.guided.dtable.client.handlers;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import com.google.gwt.dom.client.AnchorElement;
+import com.google.gwt.dom.client.Document;
+import com.google.gwt.dom.client.LIElement;
+import com.google.gwt.event.logical.shared.ValueChangeEvent;
+import com.google.gwt.event.logical.shared.ValueChangeHandler;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.user.client.Event;
+import com.google.gwt.user.client.ui.Composite;
+import org.drools.workbench.models.guided.dtable.shared.model.GuidedDecisionTable52;
+import org.drools.workbench.screens.guided.dtable.client.resources.HitPolicyInternationalizer;
+import org.jboss.errai.common.client.dom.Node;
+import org.jboss.errai.common.client.dom.Span;
+import org.jboss.errai.common.client.dom.UnorderedList;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+
+@Dependent
+@Templated
+public class HitPolicySelectorViewImpl
+        extends Composite
+        implements HitPolicySelectorView {
+
+    @Inject
+    @DataField("selectedValue")
+    Span selectedValue;
+
+    @Inject
+    @DataField("listOfAvailableHitModes")
+    UnorderedList listOfAvailableHitModes;
+
+    @Override
+    public void setSelection( final GuidedDecisionTable52.HitPolicy hitPolicy ) {
+        selectedValue.setTextContent( HitPolicyInternationalizer.internationalize( hitPolicy ) );
+    }
+
+    @Override
+    public void addHitPolicyOption( final GuidedDecisionTable52.HitPolicy hitPolicy ) {
+        final AnchorElement anchor = Document.get()
```

Comment:
@jomarko Late reply. 
That is the easy implementation, but setting it up gets messy unless you can use GWT Component and we are trying to get rid of GWT Components and Widgets since the future GWT versions will not support them. The new way is to use Elements. This however makes life a bit more complicated, at least right now when we are in transition phase. One example is this implementation that would have been a quick thing with the old way, but not this should probably be made into a reusable widget somewhere in our code base.